cranelift/filetests/src/runtest_environment.rs

```diff
@@ -75,6 +103,34 @@ impl RuntestEnvironment {
 
         Ok(())
     }
+
+    /// Allocates a struct to be injected into the test.
+    pub fn runtime_struct(
+        &self,
+        mut alloc_heap: impl FnMut(u64) -> u64,
+        mut alloc_table: impl FnMut(u64, u64) -> u64,
+    ) -> Vec<u64> {
+        let context_struct = self
+            .entries
+            .iter()
+            .flat_map(|entry| match entry {
+                RuntestEntry::Heap(heap) => {
+                    let size: u64 = heap.size.into();
+                    [alloc_heap(size), size]
```

Comment:
This is a behaviour change from what we had before.

We used to put the start and end pointers. However it looks like that's not whats expected by the `bound` mechanism for heaps.

This changes it to pass the size in bytes of the heap.